src/IntegratorLock/IntegratorLockCleaner.php

```diff
@@ -41,25 +46,31 @@ public function deleteLock(): void
     {
         $lockFilePath = $this->config->getIntegratorLockFilePath();
 
-        unlink($lockFilePath);
-
-        if (!$this->isLockFileChangeTrackedByGit($lockFilePath)) {
+        if ($this->isLockFileIgnoredByGit($lockFilePath) && !file_exists($lockFilePath)) {
             return;
         }
+        if (file_exists($lockFilePath)) {
+            unlink($lockFilePath);
+        }
+        $gitignorePath = $this->config->getProjectRootDirectory() . static::GITIGNORE_FILE;
+
+        if (!file_exists($gitignorePath) || strpos((string)file_get_contents($gitignorePath), $this->config::INTEGRATOR_LOCK) === false) {
```

Comment:
`strpos((string)file_get_contents($gitignorePath), $this->config::INTEGRATOR_LOCK) === false` will never be true, 'cause it casts to string

Reply:
No, it's not the same. return type for strpos false or int. everything is correct here